### PR TITLE
Cleanup docker

### PR DESCRIPTION
**Pull request by rafenden**
Cleanup our Docker setup and align the way we run tests locally and on CIrcle CI.

The new approach is taking advantage of Makefile so developers can run tests and start the app with a single command, e.g. `make start-mock` or `make unit-tests`.

No local dependencies except Docker are required.

### CODE REVIEW

File: src/client/components/Form/elements/FieldWrapper.jsx [hunk 1]
```diff
@@ -16,12 +16,11 @@ import Paragraph from '@govuk-react/paragraph'
 const StyledFormGroup = styled(FormGroup)`
   ${(props) => {
     const isReduced = props.reduced
-    const isIE = props.isIE
     const hasReducedPadding = props.reducedPadding
     const hasHint = !!props.hint
 
     return `
-    ${(isReduced || isIE) &&
+    ${isReduced &&
```
Comment (ian-leggett) [inline src/client/components/Form/elements/FieldWrapper.jsx:23]: This line seems to have been effected?

File: test/functional/cypress/specs/companies/filter-spec.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ describe('Company Collections Filter', () => {
         cy.contains('Search multiple postcodes separated by a comma')
         cy.get('input')
           .type(POSTCODE)
-          .blur()
+          .type('{enter}')
```
Comment (ian-leggett) [inline test/functional/cypress/specs/companies/filter-spec.js:176]: Same as above

File: test/functional/cypress/specs/companies/referrals/referral-help-spec.js [hunk 3]
```diff
@@ -49,11 +49,12 @@ describe('Referral help', () => {
...
-          `Forward this referral onto someone else. Paste this URL into an email: http://localhost:3000${urls.companies.referrals.details(
-            companyId,
-            '1'
-          )}`
+          'contain.text',
+          'Forward this referral onto someone else. Paste this URL into an email:'
+        )
+        .should(
+          'contain.text',
+          `${urls.companies.referrals.details(companyId, '1')}`
```
Comment (ian-leggett) [inline test/functional/cypress/specs/companies/referrals/referral-help-spec.js:57]: As above

File: test/functional/cypress/specs/companies/filter-spec.js (hunk 2 above)
Comment (rafenden) [reply test/functional/cypress/specs/companies/filter-spec.js:176]: Some tests were flaky and had to be fixed.

File: test/functional/cypress/specs/companies/add-company-spec.js [hunk 4]
```diff
@@ -591,7 +591,7 @@ describe('Add company form', () => {
     })
 
     it('should render an "Add a company" H1 element', () => {
-      cy.get(selectors.companyAdd.title).should('have.prop', 'tagName', 'H1')
+      cy.get(selectors.companyAdd.title).should('have.text', 'Add company')
```
[the remark this answers is not in the document]
Comment (rafenden) [reply test/functional/cypress/specs/companies/add-company-spec.js:594]: Some tests were flaky and had to be fixed.

File: src/client/components/Form/elements/FieldWrapper.jsx (hunk 1 above)
Comment (rafenden) [reply src/client/components/Form/elements/FieldWrapper.jsx:23]: Good catch, it's now reversed.

File: test/functional/cypress/specs/companies/referrals/referral-help-spec.js (hunk 3 above)
Comment (rafenden) [reply test/functional/cypress/specs/companies/referrals/referral-help-spec.js:57]: This test was flaky so it had to be fixed.

File: .bin.sample/start-dev-api [hunk 5]
```diff
@@ -0,0 +1 @@
+API_ROOT=https://dev.api.datahub.example.com OAUTH2_DEV_TOKEN=userToken yarn docker:base:start
```
[the remark this answers is not in the document]
Comment (debitan) [reply .bin.sample/start-dev-api:1]: also, I can't see this script in the package.json

[the remark this answers is not in the document]
Comment (rafenden) [reply .bin.sample/start-dev-api]: This was an outdated file and has been removed now.

File: .bin.sample/start-staging-api [hunk 6]
```diff
@@ -0,0 +1 @@
+API_ROOT=https://staging.api.datahub.example.com OAUTH2_DEV_TOKEN=userToken yarn docker:base:start
```
[the remark this answers is not in the document]
Comment (rafenden) [reply .bin.sample/start-staging-api]: This was an outdated file and has been removed now.

File: package.json [hunk 7]
```diff
@@ -8,33 +8,30 @@
...
-    "test:e2e:da": "CYPRESS_integrationFolder=test/end-to-end/cypress/specs/DA cypress run --config baseUrl=http://localhost:3000",
-    "test:e2e:dit": "CYPRESS_integrationFolder=test/end-to-end/cypress/specs/DIT cypress run --config baseUrl=http://localhost:3000",
-    "test:e2e:dit:watch": "CYPRESS_integrationFolder=test/end-to-end/cypress/specs/DIT cypress open --config baseUrl=http://localhost:3000",
-    "test:e2e:watch": "CYPRESS_integrationFolder=test/end-to-end/cypress/specs cypress open --config baseUrl=http://localhost:3000",
-    "test:visual": "./node_modules/.bin/wdio test/visual/wdio.conf.js",
+    "test:e2e:lep": "CYPRESS_coverage=false CYPRESS_integrationFolder=test/end-to-end/cypress/specs/LEP cypress run --browser chrome",
+    "test:e2e:da": "CYPRESS_coverage=false CYPRESS_integrationFolder=test/end-to-end/cypress/specs/DA cypress run --browser chrome",
+    "test:e2e:dit": "CYPRESS_coverage=false CYPRESS_integrationFolder=test/end-to-end/cypress/specs/DIT cypress run --browser chrome",
+    "test:e2e:watch": "CYPRESS_coverage=false CYPRESS_integrationFolder=test/end-to-end/cypress/specs cypress open",
+    "test:visual": "npx wdio test/visual/wdio.conf.js",
```
[the remark this answers is not in the document]
Comment (rafenden) [reply package.json:31]: You can :)

Just run `make visual-tests` and you are good to go.

**Pull request merged**